contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -0,0 +1,410 @@
+        $publicParameters = $PSBoundParameters
+        $publicParameters["FilePath"] += ".pub"
+        Fix-FilePermissions -Owners $currentUser, $adminsAccount,$systemAccount -AnyAccessOK $currentUser -ReadAccessOK $everyone @publicParameters
+}
+
+<#
+    .Synopsis
+    Fix-UserSSHConfigPermissions
+    fix the file owner and permissions of user config
+#>
+function Fix-UserSSHConfigPermissions
+{
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [switch] $Quiet)
+        Fix-FilePermissions -Owners $currentUser,$adminsAccount,$systemAccount -AnyAccessOK $currentUser @psBoundParameters
+}
+
+<#
+    .Synopsis
+    Fix-FilePermissionInternal
+    Only validate owner and ACEs of the file
+#>
+function Fix-FilePermissions
+{
+    param (        
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [ValidateNotNull()]
+        [System.Security.Principal.NTAccount[]] $Owners = $currentUser,
+        [System.Security.Principal.NTAccount[]] $AnyAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessNeeded,
+        [switch] $Quiet
+    )   
+    
+    Write-host "----------Validating $FilePath----------"
+    $return = Fix-FilePermissionInternal @PSBoundParameters
+
+    if($return -contains $true) 
+    {
+        #Write-host "Re-check the health of file $FilePath"
+        Fix-FilePermissionInternal @PSBoundParameters
+    }
+}
+
+<#
+    .Synopsis
+    Fix-FilePermissionInternal
+#>
+function Fix-FilePermissionInternal {
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$FilePath,
+        [ValidateNotNull()]
+        [System.Security.Principal.NTAccount[]] $Owners = $currentUser,
+        [System.Security.Principal.NTAccount[]] $AnyAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessNeeded,
+        [switch] $Quiet
+    )
+
+    $acl = Get-Acl $FilePath
+    $needChange = $false
+    $health = $true
+    if ($Quiet)
+    {
+        $result = 'Y'
+    }
+    
+    if(-not $Owners.Contains([System.Security.Principal.NTAccount]$($acl.Owner)))
+    {
+        if (-not $Quiet) {
+            $warning = "Current owner: '$($acl.Owner)'. '$($Owners[0])' should own $FilePath."
+            Do {
+                Write-Warning $warning
+                $input = Read-Host -Prompt "Shall I set the file owner? [Yes] Y; [No] N (default is `"Y`")"
+                if([string]::IsNullOrEmpty($input))
+                {
+                    $input = 'Y'
+                }        
+            } until ($input -match "^(y(es)?|N(o)?)$")
```

Comment:
not sure if its case sensitive here.. if yes then change "y" to "Y" to be consistent with line 167